onnx/onnxifi_dummy.c

```diff
@@ -174,10 +174,10 @@ onnxGetExtensionFunctionAddress(
     if (strcmp(name, extension_function_list[i]) == 0) {
       switch (i) {
         case 0:
-          *function = &onnxGetExtensionFunctionAddress;
+          *function = (void *)&onnxGetExtensionFunctionAddress;
```

Comment:
Why is this "void *" ? Why not "onnxExtensionFunctionPointer", which is the declared type of *function?

Reply:
Because that would produce another warning due to the different function signature.
```
[ 23%] Building C object third_party/onnx/CMakeFiles/onnxifi_dummy.dir/onnx/onnxifi_dummy.c.o
/root/Downloads/onnx-mlir/third_party/onnx/onnx/onnxifi_dummy.c: In function ‘onnxGetExtensionFunctionAddress’:
/root/Downloads/onnx-mlir/third_party/onnx/onnx/onnxifi_dummy.c:177:23: warning: cast between incompatible function types from ‘onnxStatus (*)(void *, const char *, onnxStatus (**)(void))’ {aka ‘int (*)(void *, const char *, int (**)(void))’} to ‘onnxStatus (*)(void)’ {aka ‘int (*)(void)’} [-Wcast-function-type]
  177 |           *function = (onnxExtensionFunctionPointer)&onnxGetExtensionFunctionAddress;
      |                       ^
/root/Downloads/onnx-mlir/third_party/onnx/onnx/onnxifi_dummy.c:180:23: warning: cast between incompatible function types from ‘onnxStatus (*)(void *, uint32_t,  const onnxTensorDescriptorV1 *, uint32_t,  const onnxTensorDescriptorV1 *, onnxMemoryFenceV1 *)’ {aka ‘int (*)(void *, unsigned int,  const onnxTensorDescriptorV1 *, unsigned int,  const onnxTensorDescriptorV1 *, onnxMemoryFenceV1 *)’} to ‘onnxStatus (*)(void)’ {aka ‘int (*)(void)’} [-Wcast-function-type]
  180 |           *function = (onnxExtensionFunctionPointer)&onnxSetIOAndRunGraph;
      |                       ^
```